docs/local-dev.md

```diff
@@ -0,0 +1,109 @@
+# Setting up the application for local development
+
+You may run the application on a `Kubernetes cluster` whilst doing local
+developement or run the services of the application locally as well.
+
+### Prerequisites
+
+- [gcloud CLI](https://cloud.google.com/sdk/docs/install)
+- [kubectl CLI](https://kubernetes.io/docs/tasks/tools/)
+- [skaffold CLI](https://skaffold.dev/docs/install/)
+- A Google Cloud Project
+- Java **(Version 11)**
+- Node **(Version >=16.13.2, <17)**
+
+---
+### Clone the repo locally
+
+```sh
+git clone https://github.com/GoogleCloudPlatform/point-of-sale.git
+cd point-of-sale
+```
+
+### Options available for local development:
+- [Local dev with changes continuously deployed to a Kubernetes cluster](#local-development-whilst-running-the-app-in-a-k8s-cluster)
+
+- [Local dev with the entire application running in the local machine](#local-development-whilst-running-everything-locally)
+
+- [Hybrid setup (specifically for UI development)](#hybrid-setup-specifically-for-ui-development)
+---
+
+### Local development whilst running the app in a K8s cluster
+
+We use [**skaffold**](https://skaffold.dev) to automatically re-deploy the local
+changes to the cluster as we continue to update the application. Skaffold
+monitors for file changes locally and updates the cluster when it detects one.
+This option for local development is the most easiest to setup and get started.
+However, redeploying to the cluster with every change made can be a time
+consuming process.
+
+#### Steps
+- If you don't have a cluster, then create one by following the [quickstart steps](/docs/quickstart.md)
+- Make sure you have the local `kubernetes context` pointing to the cluster
```

Comment:
super small nit : maybe the cluster and the `kubectl context` bullet points should be moved to under  the Prerequisites? 🤔

Reply:
That's a good point. However, I see that it is specific to just 1/3 different local dev setups listed. Also it's less of a resource pre-req _(have skaffold installed, have a GCP project)_ and more of a _user ensuring they have the correct config_. So that's why I put it as part of the steps rather than the pre-req. :)

Thank you for the quick review. I am going to keep this as is and merge it in 🙏🏼